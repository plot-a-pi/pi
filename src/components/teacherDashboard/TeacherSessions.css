.TeacherSessions {
    font-family: 'Roboto Mono';
    width: 95vw;
    margin: auto;
    margin-top: 2vh;
    padding-top: 2vh;
    border-top:  1px solid rgba(0,0,0,.25);

    .sessionForm {
        display: flex;
        justify-content: space-evenly;
    }

    ul {
        width: 90vw;
        margin: auto;
        border-top: 1px solid rgba(0,0,0,.25);
        margin-top: 2vh;
    }

    li {
        display: block;
        margin: 1em 0 1em 0;
        padding: .45em;
        border-radius: .5em;
        box-shadow: 0 1px 0 3px rgba(0,0,0,.25);
    }

    .sessionLinks {
        display: flex;
        justify-content: space-evenly;
        margin-bottom: 2vh
    }

    h2 {
        margin-top: .75em;
    }

    h3 {
        margin-bottom: .75em;
    }

    a {
        text-decoration: none;
        background-color: transparent;
        color: whitesmoke;
     }

     button {
        min-height: 30px;
    }

    input {
        font-size: 16px;
        font-family: 'Heebo';
        color: #212E59;
        line-height: 1;
        width: 50vw;
        height: 30px;
        padding: .25em 1em .4em .5em;
        box-sizing: border-box;
        border: 1px solid #A4A6C9;
        border-radius: .5rem;
        box-shadow: 0 1px 0 1px rgba(0,0,0,.04);
        border-radius: .5em;
        -moz-appearance: none;
        -webkit-appearance: none;
        appearance: none;
        background-color: #F5F5F5;
        background-repeat: no-repeat, repeat;
        background-position: right .7em top 50%, 0 0;
        background-size: .5em auto, 100%;
    }

    input:focus {
        border-color: #aaa;
        box-shadow: 0 0 1px 2px #A4A6C9;
        box-shadow: 0 0 0 2px -moz-mac-focusring;
        color: #223493;
        outline: none;
    }
}