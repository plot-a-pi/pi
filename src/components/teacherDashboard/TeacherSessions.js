import React, { useState, useEffect } from 'react';
import { useEmitEvent, useSocketState, useSocket } from 'react-socket-io-hooks';
import { Link } from 'react-router-dom';
import { useUser } from '../../firebase/AuthProvider';
import { signOut } from '../../firebase/firebase';
import { CSVDownload } from 'react-csv';
import styles from './TeacherSessions.css';
import getSessionData from '../../services/getSessionData';


const TeacherSessions = () => {
  const [sessionName, setSessionName] = useState('Session Name');
  const [downloadData, setDownloadData] = useState(null);
  const emitUserSessions = useEmitEvent('USER_LOGIN');
  const emitNewSession = useEmitEvent('CREATE_SESSION');
  const emitRetrieveSessions = useEmitEvent('RETRIEVE_SESSIONS');
  const socket = useSocket();
  const user = useUser();

  let userName = user.displayName.split(' ')[0];


  const { sessions } = useSocketState();
  let text;

  useEffect(() => {
    if(socket.connected !== undefined) {
      emitUserSessions(user.uid);
      emitRetrieveSessions(user.uid);
    }
    if(sessions.length > 1){
      text = 'Your Saved Sessions';
    }
  }, [socket.connected]);



  const onSubmit = (event) => {
    event.preventDefault();
    emitNewSession({
      teacherId: user.uid,
      name: sessionName
    });
  };

  const handleDownload = (id) => {
    getSessionData(id)
      .then(data => data.json())
      .then(data => setDownloadData(data));
  };

  const handleClick = () => {
    signOut();
  };

  const headers = ['diameter (in)', 'circumference (in)'];

  const sessionElements = sessions.map(session => {
    return (
      <li key={session._id}>
        <h3>{session.name}</h3>
        <div className={styles.sessionLinks}>
          <Link className={styles.sessionButton} to={`/session/${session._id}`}>Get  Link</Link>
          <Link className={styles.sessionButton} to={`/session-graph/${session._id}`}>View Graph</Link>
          <Link className={styles.sessionButton} onClick={() => handleDownload(session._id)}>Download Data</Link>
        </div>
      </li>
    );
  });

  return (
    <section className={styles.wrapper}>
      <div className={styles.TeacherSessions}>
        <form onSubmit={onSubmit} className={styles.sessionForm}>
          <input type='text' placeholder={'Enter Session Name'} onChange={({ target }) => setSessionName(target.value)}></input>
          <button>Create</button>
        </form>
        <ul className={styles.sessionListWrapper}>
          <h2>{text}</h2>
          {sessionElements}
        </ul>
      </div>
      {downloadData ? <CSVDownload data={downloadData} headers={headers} /> : null}
      <div className={styles.signout}>
        <p>Signed in as: {userName}</p>
        <Link to='/'>
          <button onClick={handleClick}>Sign Out</button>
        </Link>
      </div>
    </section>
  );
};

export default TeacherSessions;
