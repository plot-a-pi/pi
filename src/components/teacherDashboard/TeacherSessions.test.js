import React from 'react';
import { shallow } from 'enzyme';
import TeacherSessions from './TeacherSessions';

jest.mock('react-socket-io-hooks', () => ({
  useEmitEvent: () => {}, 
  useSocketState: () => ({
    recentUserPoint: {},
    sessions: []
  }), 
  useSocket: () => ({
    socket: {
      connected: null
    }
  }),
}));

jest.mock('../../firebase/AuthProvider', () => ({
  useUser: () => ({
    displayName: 'test-display-name'
  }), 
}));

describe('TeacherSessions component', () => {
  it('renders the TeacherSessions', () => {
    const wrapper = shallow(<TeacherSessions />);
    expect(wrapper).toMatchSnapshot();
  });
});
