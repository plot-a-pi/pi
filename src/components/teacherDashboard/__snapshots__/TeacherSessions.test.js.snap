// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherSessions component renders the TeacherSessions 1`] = `
<section
  className="wrapper"
>
  <div
    className="TeacherSessions"
  >
    <form
      className="sessionForm"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Enter Session Name"
        type="text"
      />
      <button>
        Create
      </button>
    </form>
    <ul
      className="sessionListWrapper"
    >
      <h2 />
    </ul>
  </div>
  <div
    className="signout"
  >
    <p>
      Signed in as: 
      test-display-name
    </p>
    <Link
      to="/"
    >
      <button
        onClick={[Function]}
      >
        Sign Out
      </button>
    </Link>
  </div>
</section>
`;
